File: solutions/022/022.js
const fs = require('fs')

const names = fs.readFileSync('./resources/names.txt').toString().split(',')
  .map(x => x.replace(/\"/g, '')).sort()

let sum = 0
for (let i = 0; i < names.length; i++) {
  sum += names[i].split('').map(x => x.charCodeAt(0) - 64)
    .reduce((a, b) => a + b) * (i + 1)
}
console.log(sum)

File: solutions/022/022.go
package main

import (
	"fmt"
	"io/ioutil"
	"os"
	"sort"
	"strings"
)

func nameScore(name string) (score int) {
	for _, letter := range name {
		score += int(letter - 64)
	}
	return
}

func main() {
	file, _ := os.Open("./resources/names.txt")
	defer file.Close()
	bs, _ := ioutil.ReadAll(file)
	names := strings.Split(string(bs), ",")
	sort.Strings(names)
	sum := 0
	for position, name := range names {
		name := strings.Replace(name, "\"", "", -1)
		sum += nameScore(name) * (position + 1)
	}
	fmt.Println(sum)
}
